admin/modules/user/awaiting_activation.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * MyBB 1.8
+ * Copyright 2014 MyBB Group, All Rights Reserved
+ *
+ * Website: http://www.mybb.com
+ * License: http://www.mybb.com/about/license
+ *
+ */
+
+// Disallow direct access to this file for security reasons
+if(!defined("IN_MYBB"))
+{
+	die("Direct initialization of this file is not allowed.<br /><br />Please make sure IN_MYBB is defined.");
+}
+
+$page->add_breadcrumb_item($lang->awaiting_activation, "index.php?module=user-awaiting_activation");
+
+$sub_tabs['awaiting_activation'] = array(
+	'title' => $lang->awaiting_activation,
+	'link' => "index.php?module=user-awaiting_activation",
+	'description' => $lang->awaiting_activation_desc
+);
+
+$plugins->run_hooks("admin_user_awaiting_activation_begin");
+
+if($mybb->input['action'] == "activate" && $mybb->request_method == "post")
+{
+	$plugins->run_hooks("admin_user_awaiting_activation_activate");
+
+	$mybb->input['user'] = array_map('intval', $mybb->input['user']);
+	$user_ids = implode(", ", $mybb->input['user']);
+
+	if(empty($user_ids))
+	{
+		flash_message($lang->no_users_selected, 'error');
+		admin_redirect("index.php?module=user-awaiting_activation");
+	}
+
+	$num_activated = $num_deleted = 0;
+	if($mybb->input['delete']) // Delete selected user(s)
+	{
+		require_once MYBB_ROOT.'inc/datahandlers/user.php';
+		$userhandler = new UserDataHandler('delete');
+
+		$query = $db->simple_select("users", "uid, usergroup", "uid IN ({$user_ids})");
+		while($user = $db->fetch_array($query))
+		{
+			if($user['usergroup'] == 5)
+			{
+				++$num_deleted;
+				$userhandler->delete_user($user['uid'], 1);
```

Comment:
Probably instead of deleting every user in a separate call you should save them in an array and pass that array to `delete_user`. This would reduce the number of queries here.